tests/Doctrine/Tests/DBAL/Functional/Driver/SQLAnywhere/StatementTest.php

```diff
@@ -50,4 +51,54 @@ public function testPersistentStatement()
         self::assertTrue($prepStmt->execute(),' Statement persistent failed');
     }
 
+    /**
+     * Check Issue #2991
+     * @throws \Doctrine\DBAL\ConnectionException
+     * @throws \Doctrine\DBAL\DBALException
+     * @throws \Exception
+     */
+    public function testBindParams()
+    {
+        $params = $this->_conn->getParams();
+
+        $conn = DriverManager::getConnection($params);
+
+        $conn->connect();
+        $conn->beginTransaction();
+        try {
+            self::assertTrue($conn->isConnected(),'No SQLAnywhere-Connection established');
```

Comment:
As far as I can tell, the failure to connect will throw an exception, so this line of code will not be reached. In any event, this particular test shouldn't check connection since it's not about it.

Reply:
removed